gnome-initial-setup/pages/branding-welcome/gis-branding-welcome-page.c

```diff
@@ -0,0 +1,289 @@
+      g_warning ("Could not read file %s: %s\n", CONFIG_FILEPATH, error->message);
+      goto out;
+    }
+
+  priv = gis_branding_welcome_page_get_instance_private (page);
+  groups = g_key_file_get_groups (keyfile, &n_groups);
+  for (i = 0; i < n_groups; i++)
+    {
+      size_t len;
+
+      /* We only support "Welcome" branded pages for now */
+      if (g_strcmp0 (groups[i], "Welcome") != 0)
+        continue;
+
+      priv->title = g_key_file_get_string (keyfile, groups[i], "title", NULL);
+      if (priv->title == NULL) {
+        g_warning ("Could not find 'title' in %s\n", CONFIG_FILEPATH);
+        goto out;
+      }
+
+      priv->description = g_key_file_get_string (keyfile, groups[i], "description", NULL);
+      if (priv->description == NULL)
+        g_warning ("No description found for '%s' branding page", groups[i]);
+
+      priv->logo_path = g_key_file_get_string (keyfile, groups[i], "logo", NULL);
+      if (priv->logo_path == NULL)
+        g_warning ("Unable to get logo for '%s' branding page", groups[i]);
+
+      priv->skip_pages = g_key_file_get_string (keyfile, groups[i], "skip", NULL);
+      if (priv->skip_pages == NULL)
+        g_debug ("No pages to skip for '%s' branding page", groups[i]);
+
+      /* Only "Welcome" pages are supported for now */
+      break;
+    }
+
+  ret = TRUE;
+
+ out:
+  return ret;
+}
+
+static void
+update_branding_specific_info (GisBrandingWelcomePage *page)
+{
+  GisBrandingWelcomePagePrivate *priv = gis_branding_welcome_page_get_instance_private (page);
+  GtkWidget *opt_widget = NULL;
+
+  if (!read_config_file (page)) {
+    g_debug ("No branding configuration found");
+    return;
+  }
+
+  gtk_label_set_label (GTK_LABEL (WID("branding-title")), priv->title);
+
+  if (priv->description != NULL) {
+    opt_widget = WID("branding-text");
+    gtk_label_set_label (GTK_LABEL (opt_widget), priv->description);
+    gtk_widget_set_visible (opt_widget, TRUE);
+  }
+
+  if (priv->logo_path != NULL) {
+    opt_widget = WID("branding-logo");
+    gtk_image_set_from_file (GTK_IMAGE (opt_widget), priv->logo_path);
+    gtk_widget_set_visible (opt_widget, TRUE);
+  }
+}
+
+static void
+update_line_height_for_description (GisBrandingWelcomePage *page)
+{
+  GtkWidget *widget = NULL;
+  PangoAttrList *attr_list = NULL;
+  PangoAttribute *attr = NULL;
+
+  /* We can't set CSS's line-height property in GTK+ so we have to
+   * try to get a similar result changing the baseline with pango. */
+  attr_list = pango_attr_list_new ();
+
+  /* Rising the baseline of the 14px font (see branding-welcome-page.css)
+   * by 6px gives us a 20px line height. Note that there should be a way
+   * to get this value programatically, but all the experiments I did with
+   * Pango failed so far for some reason, so I'm hardcoding this for now. */
+  attr = pango_attr_rise_new (6 * PANGO_SCALE);
+
+  pango_attr_list_insert (attr_list,attr);
+  gtk_label_set_attributes (GTK_LABEL (WID ("branding-text")), attr_list);
```

Comment:
You should be able to do this from the builder XML; see https://git.gnome.org/browse/gtk+/tree/gtk/gtklabel.c#n106